pkg/reconciler/contour/lister.go

```diff
@@ -44,15 +44,17 @@ func (l *lister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([
 	var results []status.ProbeTarget
 
 	cfg := config.FromContext(ctx)
-
-	port, scheme := int32(80), "http"
-	switch cfg.Network.HTTPProtocol {
-	case network.HTTPDisabled, network.HTTPRedirected:
-		port, scheme = 443, "https"
-	}
-
 	visibilityKeys := cfg.Contour.VisibilityKeys
+
 	for key, hosts := range ingress.HostsPerVisibility(ing, visibilityKeys) {
+		port, scheme := int32(80), "http"
+
+		// Probe external servce with https.
+		if ing.Spec.HTTPOption == v1alpha1.HTTPOptionRedirected &&
+			!visibilityKeys["ClusterLocal"].Has(key) {
```

Comment:
cc @kvmware

I wonder if this line fixes this issue: https://github.com/knative-sandbox/net-contour/issues/423